### ISSUE TITLE
Reading multiple folders

### ISSUE THREAD

**Issue by SrikanthTati** [databricks/spark-csv#242]
Hello,

I'm trying to read multiple folders each containing several lzo compressed csv files.

    val datePath = "2016-01-22-"
    val fileList = (1 to 3).toList.map(pathPrefix+datePath+_+"/*.lzo")
    val DFLzo = sqlContext.read.format("com.databricks.spark.csv")
                      .options(Map("delimiter" -> "\t"))
                      .schema(RawLogsSchema.getSchema)
                      .load(fileList : _*)

But I get this exception. Any idea what's missing?
org.apache.spark.sql.AnalysisException: com.databricks.spark.csv.DefaultSource does not support paths option.;
        at org.apache.spark.sql.execution.datasources.ResolvedDataSource$.apply(ResolvedDataSource.scala:109)

**Comment by falaki** [comment]
You need to use a wild card path and let Spark take care of it.
```val DFLzo = sqlContext.read.format("com.databricks.spark.csv")
                  .options(Map("delimiter" -> "\t"))
                  .schema(RawLogsSchema.getSchema)
                  .load("2016-01-22-*")
```

**Comment by SrikanthTati** [comment]
This was just an example I gave. My folder structure is ../YYY-MM-DD-HH/*.lzo
For a given run I have a few days and within each day I don't always process all hours.
Ex, process 2016-01-22-18 to 2016-01-27-06 in this job. So I can't say 2016-01-22-*
Start and end day-hour is an input to my program.

**Comment by falaki** [comment]
In that case create multiple DataFrames and union them.

**Comment by SrikanthTati** [comment]
That is what I have right now. I was trying to make it elegant.

When I make multiple calls to sqlContext.read.format().load() with smaller number of files and do a union, when an action is performed on the DF, I presume DAG will place all the calls in the same stage and use same no.of tasks to read the files as it would if I call sqlContext.read.format() once rt?

**Comment by SrikanthTati** [comment]
My use case boils down to having partition by(dayHour) and then a filter(dayHour between X and Y)
My understanding was that this was not supported yet [issue 123](https://github.com/databricks/spark-csv/issues/123).
Today I noticed another old [issue](https://github.com/databricks/spark-csv/issues/50) that says partition by is available in spark-csv since 1.4 release.

Can you clarify if this is supported?
